src/net40/Radical.Windows.Presentation/Regions/Specialized/TabControlRegion.cs

```diff
@@ -39,16 +39,27 @@ public TabControlRegion( string name )
         /// <param name="view">The view.</param>
         protected override void OnAdd(DependencyObject view)
         {
-            var tabItem = new TabItem();
-            tabItem.Content = view;
+            if (view == null) throw new ArgumentNullException(nameof(view));
 
-            var header = this.TryGetHeader(view);
-            if (header != null)
+            TabItem tabItem;
+            if (view is TabItem)
             {
-                tabItem.Header = header;
+                tabItem = (TabItem) view;
             }
+            else
+            {
+                tabItem = new TabItem
+                {
+                    Content = view
+                };
 
-            this.Element.Items.Add(tabItem);
+                object header = this.TryGetHeader(view);
+                if (header != null)
+                {
+                    tabItem.Header = header;
+                }                
+                this.Element.Items.Add(tabItem);
```

Comment:
The add has to be outside of the if/else.

Reply:
Ops 😄